test/e2e/framework/service/jig.go

```diff
@@ -55,8 +57,20 @@ import (
 // NodePortRange should match whatever the default/configured range is
 var NodePortRange = utilnet.PortRange{Base: 30000, Size: 2768}
 
-// It is copied from "k8s.io/kubernetes/pkg/registry/core/service/portallocator"
-var errAllocated = errors.New("provided port is already allocated")
+// Initialize only once per test
+var (
+	staticPortAllocator = &staticPortRange{
+		nodeports: [87]bool{false},
+		baseport:  30000,
+		mu:        &sync.Mutex{},
+	}
+)
+
+type staticPortRange struct {
+	mu        *sync.Mutex
+	baseport  int
+	nodeports [87]bool // this is the static port range based on current NodePortRange KEP-3668
```

Comment:
fixed-length arrays are also generally weird and rare. And anyway, in this case you shouldn't be hardcoding 87 anyway; you should be computing both `baseport` and the length of `nodeports` from the values in `NodePortRange` above.

This allocator isn't going to be used much, and isn't going to contain many values. I think it would probably be simplest to have a `baseport` and `length`, and then a `sets.Set[int]` to keep track of the ports that are currently allocated.

Reply:
Thanks for review comment. makes sense, but in that case do we hardcode `minPort := 16` , `step := 32`, `maxPort := 128` or are they computed somehow ?

Reply:
@danwinship that var is overriden in some place?

Reply:
> in that case do we hardcode `minPort := 16` , `step := 32`, `maxPort := 128` or are they computed somehow ?

Yeah, you'd hardcode those; that's part of the definition of how the feature works.

(Maybe it would be nice to have a helper function shared between this and kube-controller-manager, but that would be a bunch of work for not much gain.)

> that var is overriden in some place?

No, but someone might patch it locally. The comment on it says "NodePortRange should match whatever the default/configured range is" so presumably the point is that if someone does need to change it, they can.

Reply:
I see, you are right,  @Anurag252 can you please follow Dan suggestion? sorry for the inconvenience